File: kora-app-annotation-processor/src/main/java/ru/tinkoff/kora/kora/app/annotation/processor/interceptor/ComponentInterceptor.java
package ru.tinkoff.kora.kora.app.annotation.processor.interceptor;

import ru.tinkoff.kora.kora.app.annotation.processor.component.ResolvedComponent;

import javax.lang.model.type.TypeMirror;

public record ComponentInterceptor(ResolvedComponent component, TypeMirror interceptType) {
}

File: kora-app-symbol-processor/src/main/kotlin/ru/tinkoff/kora/kora/app/ksp/interceptor/ComponentInterceptor.kt
package ru.tinkoff.kora.kora.app.ksp.interceptor

import com.google.devtools.ksp.symbol.KSType
import ru.tinkoff.kora.kora.app.ksp.component.ResolvedComponent
import ru.tinkoff.kora.kora.app.ksp.declaration.ComponentDeclaration

data class ComponentInterceptor(
    val component: ResolvedComponent,
    val declaration: ComponentDeclaration,
    val interceptType: KSType
)
